src/crt/sllnot.src

```diff
@@ -0,0 +1,76 @@
+; Performs 48 bit ("short long long") bitwise not
+; 
+; Arguments:
+; ude = Most significant bytes of arg 1
+; uhl = Least significant bytes of arg 1
+
+
+; Returns:
+; ude:uhl = ~ude:uhl
+
+	assume	adl=1
+
+	section	.text
+	public	__sllnot
+__sllnot:
+	push de
+	ex de, hl
+	scf
+	sbc hl, hl ; hl = -1
+	or a, a ; reset carry flag
+	sbc hl, de ; hl = -1-hl
+	push hl
+
+	; -6: ~l               <-- SP
+	; -5: ~h
+	; -4: ~u[hl]
+	; -3: e
+	; -2: d
+	; -1: u[de]
+	; -0: everything else
+	
+	inc sp
```

Comment:
This technique is invalid if interrupts are enabled. Anything below (SP) can be considered trashed at any time.

Reply:
Revised version no longer uses this technique